hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/action/commit/BucketBulkInsertDataInternalWriterHelper.java

```diff
@@ -91,27 +90,20 @@ private UTF8String extractRecordKey(InternalRow row) {
     }
   }
 
-  protected HoodieRowCreateHandle getBucketRowCreateHandle(String fileId, int bucketId) throws Exception {
-    if (!handles.containsKey(fileId)) { // if there is no handle corresponding to the fileId
```

Comment:
The fileid passed here is actually partitonPath,  so `handles.containsKey(fileId)` is always true for record in the same partition thus no new handle will be created.